RecoVertex/PrimaryVertexProducer/interface/SequentialPrimaryVertexFitterAdapter.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SequentialPrimaryVertexFitterAdapter_h
+#define SequentialPrimaryVertexFitterAdapter_h
+
+/**\class SequentialPrimaryVertexFitterAdapter
+ 
+  Description: Adapter class for Kalman and Adaptive vertex fitters 
+
+*/
+
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "RecoVertex/PrimaryVertexProducer/interface/PrimaryVertexFitterBase.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexFitter.h"
+
+class SequentialPrimaryVertexFitterAdapter : public PrimaryVertexFitterBase {
+public:
+  SequentialPrimaryVertexFitterAdapter() : fitter(nullptr){};
+  SequentialPrimaryVertexFitterAdapter(VertexFitter<5>* vertex_fitter) : fitter(vertex_fitter){};
```

Comment:
@slava77 are you implying here the use of a smart pointer according our guidelines, e.g. a shared_ptr with the calling Producer?

Reply:
looking at https://github.com/cms-sw/cmssw/blob/master/RecoVertex/VertexPrimitives/interface/VertexFitter.h all methods are const-qualified.
It would be good enough here to switch to const pointer